notebooks/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,660 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "Su5oVLIL-GK5"
+   },
+   "outputs": [],
+   "source": [
+    "import warnings\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import matplotlib.pyplot as plt\n",
+    "import seaborn as sns\n",
+    "import sklearn\n",
+    "\n",
+    "%matplotlib inline\n",
+    "warnings.simplefilter(action='ignore', category=FutureWarning)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/",
+     "height": 237
+    },
+    "id": "1jUWQVXZ-HOs",
+    "outputId": "28317993-f1b2-44c1-d468-1fd00fb58f0f",
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "train_data = pd.read_csv('C:\\\\Users\\\\Owner\\\\Desktop\\\\Proga\\\\REPOS\\\\mentorship_EPAM\\\\data\\\\train.csv', index_col = \"id\")\n",
+    "train_data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data.tail()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data.info()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data[\"family\"] = train_data[\"family\"].str.lower()     # for easier reading \n",
+    "train_data.describe(percentiles=[.25, 0.375, .5, .75, .875]).drop([\"count\"])       \n",
+    "\n",
+    "# I dropped \"count\" row to get rid of exponentioal number presentation\n",
+    "# And added .875 percentile to understand approximately how many objects have no onpromotion"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "6yDCEDYYGveo",
+    "outputId": "839affcc-2928-4892-c798-459ff56b59f0"
+   },
+   "outputs": [],
+   "source": [
+    "train_data[\"family\"].describe()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>The only categorical feature (in this table) \"family\" has 33 possible unique values (not so a lot) <br>\n",
+    "That means we can easily use one-hot-encoding during model training</i>"
```

Comment:
I would rather say 33 is quite big number: if you apply one-hot encoding, you will obtain 33 new features.
By the way, why do you think that one-hot encoding is a good option for processing this variable? Do you know any other techiques for processing categorical variables?

Reply:
I know two main ways of processing categorical variables: **label encoding** and **one-hot encoding**. I decided to choose one-hot-encoding, because in comparison with label encoding it will save interpretability of categorical features. After using label encoding we could have value "1" (for example, 'automotive' family) and value "33" (for example, 'seafood'). It means, that these categories are completely unequal, but it's wrong. Actually, I think one more technique can be applied here - target encoding. But is necessary to keep in mind, that there is a high risk of overfitting (and also the training set and the test set should be similarly distributed).

Reply:
I read about some methods for reducing the dimensionality of the table. I never met them in my practice before, but maybe one of them can be used here together with the one-hot encoding...